pkg/service/volume_service_test.go

```diff
@@ -112,6 +122,9 @@ func (suite *VolumeServiceTestSuite) createVolColl(name string) {
 }
 
 func (suite *VolumeServiceTestSuite) TestGetNonExistentVolumeByID() {
+	if suite.groupService == nil {
```

Comment:
What a bummer to put these checks everywhere.  Is there no better option?

Reply:
Agreed.  Why were they added?

It seems since you check for this SetupTest() above, it will fail there and none of these tests will run.

Reply:
It runs and panic in all the test cases . Thus had to put nil check everywhere.

Reply:
No it doesnt , it runs all the test even setup failed.

Reply:
If setup fails, with this change does the test case pass or fail?

Reply:
setup is not a test , all the test starts with Test* methods so eventually it will come to all the Test* methods even after setup has error .

Reply:
Let me check the documentation of suite package . will see if anything can be done about it .

Reply:
Found it , we just need to exit the suite. However there is no way via suite package . pls check last commit